libs/datamodel/core/tests/config/sources.rs

```diff
@@ -631,6 +631,50 @@ fn referential_integrity_default() {
     assert_eq!(config.referential_integrity(), Some(ReferentialIntegrity::ForeignKeys));
 }
 
+#[test]
+fn cockroach_provider_is_behind_preview_feature() {
+    let dm = r#"
+        datasource ps {
+          provider = "cockroachdb"
+          url = env("DATABASE_URL")
+        }
+    "#;
+
+    let error = super::parse_config(dm).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mDatasource provider not known: "cockroachdb".[0m
```

Comment:
Technically it's "known" but need the preview feature.
We could have something that say something like that in the future maybe?

Reply:
yep, it would be better ergonomics. It's not a ton of work.